File: cpp/luogu/old/P1177.cpp
/*@author    Allergy
 *@email     Allergy527@gmail.com
 *@workspace luogu\P1177.cpp
 *@data      2024/01/17 19:14:36
 */
#include<bits/stdc++.h>
#define ll long long

using namespace std;

void solve() {
    int n, i;
    cin >> n;
    vector<int>qwq(n);
    for(auto &x : qwq)cin >> x;
    sort(qwq.begin(), qwq.end());
    for(i = 0;i < n;++i)cout << qwq[i] << (i == n - 1 ? '\n' : ' ');
}

int main() {
    ios::sync_with_stdio(false);
    int t = 1;
    //cin >> t;
    while(t--)solve();
    cin >> t;
    return 0;
}

File: rust_luogu/src/bin/Part 2.2/p1177.rs
//@author    Allergy
//@email     Allergy527@gmail.com
//@workspace luogu/p1177.rs
//@data      2024/01/17 23:27:33
fn cin() -> String {
    let mut input = String::new();
    std::io::stdin().read_line(&mut input).unwrap();
    input.trim().to_string()
}
fn main() {
    let t = 1;
    //let mut input = String::new();std::io::stdin().read_line(&mut input).unwrap();let t:i32=input.parse().unwrap();
    (0..t).for_each(|_| solve());
}
fn solve() {
    cin();
    let mut qwq = cin()
        .split_whitespace()
        .map(|s| s.parse().unwrap())
        .collect::<Vec<i32>>();
    qwq.sort_unstable();
    qwq.iter()
        .enumerate()
        .for_each(|(i, x)| print!("{}{}", x, if i == qwq.len() - 1 { '\n' } else { ' ' }))
}
